aspnetcore/performance/response-compression.md

```diff
@@ -0,0 +1,198 @@
+
+The headers involved in requesting, sending, caching, and receiving compressed content are described below.
+
+Header | Role
+--- | ---
+`Accept-Encoding` | Sent by the client to the server to indicate which types of content encoding are acceptable.
+`Content-Type` | Specifies the MIME type of the content. MIME types are specified on the server for encodable content, so this information is used by the server to determine if a response can be compressed. The middleware includes a set of default MIME types that it will encode, but you can replace or add MIME types.
+`Content-Encoding` | Sent from the server to the client to indicate the encoding of the content in the payload.
+`Vary` | When sent by the server with a value of `Accept-Encoding` to clients and proxies, it indicates that they should cache (vary) responses based on the value of the `Accept-Encoding` header of the request. The result of returning content with the `Vary: Accept-Encoding` header is that both compressed and uncompressed responses will be cached.
+
+## Response compression sample application
+You can explore the features of the Response Compression Middleware with the [response compression sample application](https://github.com/aspnet/Docs/tree/master/aspnetcore/performance/response-compression/sample/FullSample). The sample illustrates the compression of application responses using GZip and custom compression providers. It also shows you how to add a MIME type to the default list of MIME types for compression.
+
+## When to use Response Compression Middleware
+Use Response Compression Middleware when you're unable to use the [Dynamic Compression module](https://www.iis.net/overview/reliability/dynamiccachingandcompression) in IIS on Windows Server, the [Apache mod_deflate module](http://httpd.apache.org/docs/current/mod/mod_deflate.html), [NGINX Compression and Decompression](https://www.nginx.com/resources/admin-guide/compression-and-decompression/), or your application is hosted on [WebListener server](xref:fundamentals/servers/weblistener). The main reason to use the server-based response compression technologies in IIS, Apache, or Nginx is that the performance of the middleware probably won't match that of the server modules. 
+
+## Package
+To include the middleware in your project, add a reference to the `Microsoft.AspNetCore.ResponseCompression` package. The middleware is available for projects that target .NETFramework 4.5.1 or .NETStandard 1.3 or higher.
+
+## Configuration
+### Using defaults
+If you plan to implement the middleware with default GZip compression and for default MIME types (listed below in the [MIME types section](#mime-types)), you can add the middleware to your service collection and processing pipeline. No additional configuration is required. 
```

Comment:
Should call out somewhere in the docs that if compression occurs, the middleware also strips out the MD5 header (since the body content changed so the hash is no longer valie) and removes the Content-Length header (again since the content of the body will change after compression)